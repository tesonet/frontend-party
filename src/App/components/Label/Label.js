import React from 'react';
import styled from 'styled-components';

const Label = ({ size }) => {
  return (
    <Container size={size}>
      testio
      <LabelDot size={size} />
    </Container>
  )
}

const Container = styled.div`
  font-size: ${props => props.size === 'large' ? '50px' : '30px'};
  display: flex;
  justify-content: center;
  font-weight: 900;
  text-align: center;
  align-items: baseline;
  margin-bottom: ${props => props.size === 'large' ? '40px' : '0'};
  color: ${props => props.size === 'large' ? 'white' : 'rgb(41, 44, 74)'};
`

const LabelDot = styled.div`
  background-color: rgb(149, 207, 45);
  margin-left: 2px;
  border-radius: 10px;
  height: ${props => props.size === 'large' ? '12px' : '7px'};
  width: ${props => props.size === 'large' ? '12px' : '7px'};
`;

export default Label;
