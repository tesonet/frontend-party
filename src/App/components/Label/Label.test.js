import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

it('renders correctly small label', () => {
  const LabelComponent = renderer
    .create(<Label size="small" />)
    .toJSON();
  expect(LabelComponent).toMatchSnapshot();
});

it('renders correctly large label', () => {
  const LabelComponent = renderer
    .create(<Label size="large" />)
    .toJSON();
  expect(LabelComponent).toMatchSnapshot();
});