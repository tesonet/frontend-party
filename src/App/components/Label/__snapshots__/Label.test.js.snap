// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly large label 1`] = `
<div
  className="sc-bdVaJa fOoPMC"
  size="large"
>
  testio
  <div
    className="sc-bwzfXH dlyMQx"
    size="large"
  />
</div>
`;

exports[`renders correctly small label 1`] = `
<div
  className="sc-bdVaJa dPZuTV"
  size="small"
>
  testio
  <div
    className="sc-bwzfXH iSHyQo"
    size="small"
  />
</div>
`;
